lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -98,7 +101,75 @@ class MockTimers {
     this.#executionQueue.removeAt(position);
   }
 
-  async * #setIntervalPromisified(interval, startTime, options) {
+  #createDate() {
+    kMock = Symbol('MockTimers');
+    /**
+     * Function to mock the Date constructor, treats cases as per ECMA-262
+     * and returns a Date object with a mocked implementation
+     * @typedef {Date} MockDate
+     * @returns {MockDate} a mocked Date object
+     */
+    function MockDate(year, month, date, hours, minutes, seconds, ms) {
+      const mockTimersSource = MockDate[kMock];
+      const nativeDate = mockTimersSource.#realDate;
+      // As of the fake-timers implementation for Sinon
+      // ref https://github.com/sinonjs/fake-timers/blob/a4c757f80840829e45e0852ea1b17d87a998388e/src/fake-timers-src.js#L456
+      // This covers the Date constructor called as a function ref.
+      // ECMA-262 Edition 5.1 section 15.9.2.
+      // and ECMA-262 Edition 14 Section 21.4.2.1
+      if (!(this instanceof MockDate)) {
+        return new nativeDate(mockTimersSource.#now).toString();
+      }
+
+
+
+      // Cases where Date is called as a constructor
+      switch (arguments.length) {
+        case 0:
+          return new nativeDate(MockDate[kMock].#now);
+        case 1:
+          return new nativeDate(year);
+        case 2:
+          return new nativeDate(year, month);
+        case 3:
+          return new nativeDate(year, month, date);
+        case 4:
+          return new nativeDate(year, month, date, hours);
+        case 5:
+          return new nativeDate(year, month, date, hours, minutes);
+        case 6:
+          return new nativeDate(year, month, date, hours, minutes, seconds);
+        default:
+          return new nativeDate(year, month, date, hours, minutes, seconds, ms);
+      }
+    }
+
+    // Binds all the properties of Date to the MockDate function
+    for (const prop in this.#realDate) {
+      if (ObjectPrototypeHasOwnProperty(this.#realDate, prop)) {
+        MockDate[prop] = this.#realDate[prop];
+      }
+    }
```

Comment:
note that this doesn't cover symbols or nonenumerables, nor getters/setters.

better would be:
```suggestion
    // Binds all the properties of Date to the MockDate function
    ObjectDefineProperties(
        MockDate,
        ObjectGetOwnPropertyDescriptors(this.#realDate)
    );
```

Reply:
Wow nice solution, didn't know this was a function! Thanks for the suggestion